### PR TITLE
linux-cachyos: Move AutoFDO Option to default kernel

### PR DESCRIPTION
**Pull request by ptr1337**
The last week's testing showed that the AutoFDO Integration did not provide any issues.
Lets move it to the default Kernel now, to provide this performance benefit OOB.

Until the LLVM 19 push into the archlinux repository we will use [cachyos-benchmarker](https://github.com/CachyOS/cachyos-benchmarker) to profile the kernel.

In the future, we can think about a general workflow to gather more profiles and merge these.

Propellor will be also available with LLVM 19, but for this I would like to do again a testing kernel first.

AutoFDO and Propellor has been merged into the upstream kernel, which makes it even more confident to use this feature OOB.

Closes: https://github.com/CachyOS/linux-cachyos/issues/325

### CODE REVIEW

File: linux-cachyos/PKGBUILD [hunk 1]
```diff
@@ -141,9 +141,27 @@ _build_nvidia_open=${_build_nvidia_open-}
...
+# https://cachyos.org/blog/2411-kernel-autofdo/
+# 1. Compile Kernel with _autofdo=y and _build_debug
+# 2. Boot the kernel in QEMU or on your system, see Workload
+# 3. Profile the kernel and convert the profile, see Generating the Profile for AutoFDO
+# 4. Put the profile into the sourcedir and name the profile "perf.afdo" (You can also change the name in _autofdo_profile_name)
+# 5. Enable _autofdo_use
+_autofdo=${_autofdo-}
+
+# Enable after you have put the profile into the sourcedir
+_autofdo_use=${_autofdo_use-}
```
Comment (ventureoo) [inline linux-cachyos/PKGBUILD:156]: Can't we just check availability of a profile file?

Comment (ptr1337) [reply linux-cachyos/PKGBUILD:156]: I mean, if that works - sure. Feel free to push it, if its working properly.

Comment (ventureoo) [reply linux-cachyos/PKGBUILD]: Seems to be working, but I haven't tested a full rebuild.

**Review by sirlucjan — APPROVED** [review]

File: linux-cachyos/PKGBUILD [hunk 2]
```diff
@@ -141,9 +141,24 @@ _build_nvidia_open=${_build_nvidia_open-}
...
+_autofdo=${_autofdo-}
+
+# Name for the AutoFDO profile
+_autofdo_profile_name=${_autofdo_profile_name-}
+
+
 # ATTENTION: Do not modify after this line
 _is_clang_kernel() {
-    [[ "$_use_llvm_lto" = "thin" || "$_use_llvm_lto" = "full" ]] || [ -n "$_use_kcfi" ]
+    [[ "$_use_llvm_lto" = "thin" || "$_use_llvm_lto" = "full" ]] || [ -n "$_use_kcfi" ] || [ -n "$_autofdo" ] || [ -n "$_autofdo_use" ]
```
Comment (1Naim) [inline linux-cachyos/PKGBUILD:161]:
```suggestion
    [[ "$_use_llvm_lto" = "thin" || "$_use_llvm_lto" = "full" ]] || [ -n "$_use_kcfi" ] || [ -n "$_autofdo" ]
```

File: linux-cachyos/PKGBUILD [hunk 3]
```diff
@@ -237,6 +252,15 @@ if [ -n "$_build_nvidia_open" ]; then
              "${_patchsource}/misc/nvidia/0006-nvidia-drm-Set-FOP_UNSIGNED_OFFSET-for-nv_drm_fops.f.patch")
 fi
 
+# Use generated AutoFDO Profile
+if [ -n "$_autofdo" ] && [ -n "$_autofdo_profile_name" ]; then
+    if [ -e "$_autofdo_profile_name" ]; then
+        source+=("$_autofdo_profile_name")
+    else
+        _die "You have specified _autofdo_profile_name, but file it refers to doesn't exist."
```
Comment (1Naim) [inline linux-cachyos/PKGBUILD:260]:
```suggestion
        _die "Failed to find file $_autofdo_profile_name."
```
Wonder if this would be better?

**Review by 1Naim — APPROVED** [review]
LGTM

**Comment by ptr1337** [comment]
Lets merge for 6.12.2 🎉

**Pull request merged**